src/CustomHooks/useFetch.jsx

```diff
@@ -0,0 +1,23 @@
+import { useState } from "react";
+import axios from "axios";
+
+export const useFetch = () => {
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+//   console.log("Data", data)
+  const serverCall = async (params) => {
+    setLoading(true);
+    setError(null);
+    try {
+      const res = await axios.request(params);
```

Comment:
hey, you can directly access the data here and like this const `{data} = Axios request` and also you can check if the status is 200 and then do this else do something else you can also destruct the status like this const `{data, status} = some API calls`.

Reply:
it is there in try and catch block so mostly if there is an error it will go to catch block.